server/src/main/java/io/druid/server/router/AvaticaConnectionBalancer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.router;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.druid.client.selector.Server;
+
+import java.util.Collection;
+
+/**
+ * An AvaticaConnectionBalancer balances Avatica connections across a collection of servers.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = RendezvousHashAvaticaConnectionBalancer.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "rendezvousHash", value = RendezvousHashAvaticaConnectionBalancer.class),
+    @JsonSubTypes.Type(name = "consistentHash", value = ConsistentHashAvaticaConnectionBalancer.class)
+})
+public interface AvaticaConnectionBalancer
+{
+  /**
+   * @param servers Servers to balance across
+   * @param connectionId Connection ID to be balanced
+   * @return Server that connectionId should be assigned to
+   */
+  Server balance(Collection<Server> servers, String connectionId);
```

Comment:
nit: should we just call it `pickServer(..)`  and also put in javadocs that it needs to be deterministic/sticky.

Reply:
renamed to pickServer, added comments about determinism/stickiness